traits/tests/check_observe_timing.py

```diff
@@ -0,0 +1,568 @@
+    """ Time the cases described by the (cached) property depending on an
+    extended trait scenario. Wether or not the property is cached is determined
+    by the cached_property argument, and the given property_args argument is
+    used for the Property trait defintion.
+
+    Parameters
+    ----------
+    property_args - Str
+        The string defining the argument to be passed in the definition of the
+        Property trait.  e.g. "depends_on='child.name'"
+    cached_property - Str
+        The string that will be used to decorate the getter method of the
+        Property.  Expected to be either '' or '@cached_property'.
+
+    Returns
+    -------
+    tuple
+        A 4-tuple containing the time to construct the HasTraits subclass, the
+        time to instantiate it, the time to reassign child, and the time to
+        reassign child.name
+    """
+    construct_parent_with_property = \
+        PARENT_WITH_PROPERTY_CONSTRUCTION_TEMPLATE.format(
+            property_args,
+            cached_property
+        )
+    construction_time = timeit.timeit(
+        stmt=construct_parent_with_property,
+        setup=CONSTRUCT_PARENT_SETUP,
+        number=N
+    )
+
+    instantiation_time = timeit.timeit(
+        stmt=INSTANTIATE_PARENT,
+        setup=CONSTRUCT_PARENT_SETUP + construct_parent_with_property,
+        number=N
+    )
+
+    reassign_child_time = timeit.timeit(
+        stmt=REASSIGN_CHILD,
+        setup=(
+            CONSTRUCT_PARENT_SETUP
+            + construct_parent_with_property
+            + INSTANTIATE_PARENT
+        ),
+        number=N
+    )
+
+    reassign_child_name_time = timeit.timeit(
+        stmt=REASSIGN_CHILD_NAME,
+        setup=(
+            CONSTRUCT_PARENT_SETUP
+            + construct_parent_with_property
+            + INSTANTIATE_PARENT
+        ),
+        number=N
+    )
+
+    return (
+        construction_time,
+        instantiation_time,
+        reassign_child_time,
+        reassign_child_name_time
+    )
+
+
+def report(description, benchmark_template, get_time, get_time_args):
+    """ Prints a readable benchmark report.
+
+    Parameters
+    ----------
+    descritption : str
+        The description of the benchmarking scenario being reported.
+    benchmark_template : str
+        The format string used to print the times for the benchmark in a clean,
+        formatted way
+    get_time : function
+        The function used to get the benchmark times for the current benchmark
+        scenario
+    get_time_args : list of tuples
+        The list of tuples containing the arguments to be passed the the
+        get_time function.  Note the first argument should give specifics about
+        the case being timed and will be printed as part of the report.  e.g.
+        ("@observe('name')"), or ("depends_on='name'", '@cached_property').  A
+        first argument of '' will be printed in the report as 'control'. The
```

Comment:
Instead of saying `""` indicates control (it is a bit of a magical value), we can say the first item in the list will be used as reference / control. The use of first is still a bit magical, but it will at least decouple the concept of 'control' with the specific argument values.